rwkv.cpp

```diff
@@ -600,6 +605,45 @@ struct rwkv_context * rwkv_init_from_file(const char * file_path, const uint32_t
         RWKV_ASSERT_NULL(RWKV_ERROR_MODEL_PARAMS, set_block_parameter(&parameters, i, "ffn.receptance.weight", &layer->ffn_receptance));
     }
 
+#ifdef GGML_USE_CUBLAS
+    {
+        const int n_gpu = std::min(n_gpu_layers, model->n_layer);
+
+        fprintf(stdout, "%s: [cuBLAS] offloading %d layers to GPU\n", __func__, n_gpu);
```

Comment:
I think the library should not write anything to stdout/stderr. Either a proper logging framework needs to be added (and it will not), or the library should be silent and leave actual logging to application code. Or else it will interfere with application output, which, for example, may be a chat bot interface.

Reply:
> I think the library should not write anything to stdout/stderr.

agreed

> Either a proper logging framework needs to be added (and it will not), or the library should be silent and leave actual logging to application code

I like using this callbacks for this, I use a C++ lambda callback in my newest version of some code, but there is a way to do it in pure C. For example the Lua scripting language does this

Reply:
I sotred the gpu_layer and vram_total in context, for cpp developers they could easily get the results